discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/OfferingManager.java

```diff
@@ -283,5 +286,48 @@ public String getSingleOffering() {
 
         return ret;
     }
+
+    public void initializeFromRemote(String remoteInitializationPath) {
+        try {
+            URL url = new URL(remoteInitializationPath);
+            ZipInputStream zin = new ZipInputStream(url.openStream());
+
+            /* Zip file has been downloaded, clear curreny repository directory to fill it with new information */
+            this.emptyRepository();
+            ZipEntry zipEntry = null;
+
+            File metaDirectory = this.getMetaDirectory();
+            File offeringDirectory = this.getOfferingDirectory();
+
+            while ((zipEntry = zin.getNextEntry()) != null) {
+                if (!zipEntry.isDirectory()) { /* Taking only files */
+                    String filePath = zipEntry.getName();
+                    String fileName = filePath.substring(filePath.lastIndexOf(File.separatorChar) + 1);
+                    if (filePath.startsWith("meta_directory/")) { /* meta file (.json) */
+                        this.extractFile(zin, fileName, metaDirectory);
+                    } else if (filePath.startsWith("offering_directory/")) { /* offering file (.yaml) */
+                        this.extractFile(zin, fileName, offeringDirectory);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void extractFile(ZipInputStream zin, String fileName, File outdir) {
```

Comment:
I would recommend you to use this methods instead of reinventing the wheel :)

https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#copyInputStreamToFile(java.io.InputStream, java.io.File)